### PR TITLE
feat: add metadata as optional in ReleaseProps [SPA-874]

### PR DESCRIPTION
**Pull request by Inbal-Gordon**
## Summary

The metadata prop is optional, ReleaseProps definition includes the metadata attribute

### CODE REVIEW

File: lib/entities/release.ts [hunk 1]
```diff
@@ -80,12 +81,24 @@ export type ReleaseSysProps = {
...
+      }
+    ]
+  }
+}
 /** The object returned by the Releases API */
 export interface ReleaseProps {
   title: string
   sys: ReleaseSysProps
   entities: BaseCollection<Link<Entity>>
+  metadata: MetadataAggeratedRootProps
```
Comment (anwaar931) [inline lib/entities/release.ts:101]: if metadata is optional, it should be like  `metadata?: MetadataAggeratedRootProps`

File: lib/entities/release.ts [hunk 2]
```diff
@@ -80,12 +81,24 @@ export type ReleaseSysProps = {
   updatedAt: ISO8601Timestamp
   lastAction?: Link<'ReleaseAction'>
 }
-
+export type ReleasesReferenceFilters = ScheduledActionReferenceFilters
```
Comment (anwaar931) [inline lib/entities/release.ts:84]:
```suggestion
export type ReleaseReferenceFilters = ScheduledActionReferenceFilters
```

File: lib/entities/release.ts [hunk 3]
```diff
@@ -80,12 +81,24 @@ export type ReleaseSysProps = {
...
+export type MetadataAggeratedRootProps = {
+  metadata: {
+    withReferences: [
+      {
+        entity: Link<'Entry'>
+        filter: Record<ScheduledActionReferenceFilters, string[]>[]
+      }
+    ]
+  }
+}
```
Comment (anwaar931) [inline lib/entities/release.ts:95]:
```suggestion
export type MetadataAggeratedRootProps = {
    withReferences: [
      {
        entity: Link<'Entry'>
        filter: Record<ScheduledActionReferenceFilters, string[]>[]
      }
    ]
}
```

delete `metadata` object as its already there in `ReleaseProps`

**Review by anwaar931 — CHANGES_REQUESTED** [review]

File: lib/entities/release.ts [hunk 4]
```diff
@@ -80,12 +81,24 @@ export type ReleaseSysProps = {
   updatedAt: ISO8601Timestamp
   lastAction?: Link<'ReleaseAction'>
 }
-
+export type ReleasesReferenceFilters = ScheduledActionReferenceFilters
+
+export type MetadataAggeratedRootProps = {
```
Comment (anwaar931) [inline lib/entities/release.ts:86]: I would prefer it as `ReleaseMetdata` and not link this to Aggregated Root.

File: lib/entities/release.ts (hunk 3 above)
Comment (Inbal-Gordon) [reply lib/entities/release.ts:95]: okay will do thanks!

File: lib/entities/release.ts [hunk 5]
```diff
@@ -80,12 +81,24 @@ export type ReleaseSysProps = {
...
+  metadata: {
+    withReferences: [
+      {
+        entity: Link<'Entry'>
+        filter: Record<ScheduledActionReferenceFilters, string[]>[]
```
Comment (anwaar931) [inline lib/entities/release.ts:91]: As we have created a new type of `ReleaseReferenceFilters` we should use that here.

**Review by anwaar931 — CHANGES_REQUESTED** [review]

**Comment by anwaar931** [comment]
Just some minor comments. I don't see any other issue. Maybe @SofiaMargariti can double check

File: lib/entities/release.ts (hunk 4 above)
Comment (Inbal-Gordon) [reply lib/entities/release.ts]: okie

File: lib/entities/release.ts [hunk 6]
```diff
@@ -80,12 +81,22 @@ export type ReleaseSysProps = {
   updatedAt: ISO8601Timestamp
   lastAction?: Link<'ReleaseAction'>
 }
+export type ReleaseReferenceFilters = ScheduledActionReferenceFilters
 
+export type ReleaseMetdata = {
+  withReferences: [
+    {
+      entity: Link<'Entry'>
+      filter: Record<ScheduledActionReferenceFilters, string[]>[]
```
Comment (SofiaMargariti) [inline lib/entities/release.ts:90]:
```suggestion
      filter: Record<ReleaseReferenceFilters, string[]>[]
```

File: lib/entities/release.ts [hunk 7]
```diff
@@ -80,12 +81,22 @@ export type ReleaseSysProps = {
   updatedAt: ISO8601Timestamp
   lastAction?: Link<'ReleaseAction'>
 }
+export type ReleaseReferenceFilters = ScheduledActionReferenceFilters
 
+export type ReleaseMetdata = {
```
Comment (anwaar931) [inline lib/entities/release.ts:86]: Sorry there is a type, its `ReleaseMetadata`

**Review by anwaar931 — CHANGES_REQUESTED** [review]

File: lib/entities/release.ts (hunk 6 above)
Comment (Inbal-Gordon) [reply lib/entities/release.ts:90]: oh okay

**Review by anwaar931 — APPROVED** [review]

**Pull request merged**